### ISSUE TITLE
database rollback support for the testsuite

### ISSUE THREAD

**Issue by Treggats** [tempestphp/tempest-framework#340]
**Description**
We've added support for multiple database dialects. However, this CI still only runs on SQLite.
Which is fine at first, but if we want to fully support multiple database dialects then we need to use them in the CI as well.

Setting up the CI to use those can be done pretty easily, but the testsuite at the moment isn't ready for this.

**The issue**
When running tests with SQLite I believe that the database is fully destroyed and created again. This is, however, not the case for the other dialects.
With which we run into test failures, because the table it is trying to create already exists from a previous test.

**Comment by brendt** [comment]
Why don't we spin up an individual action using a matrix? That way we always start from a clean slate?

**Comment by Treggats** [comment]
The different databases aren't the issue. Use the MySQL driver for the tests are run the testsuite. That bests demonstrates the issue

**Comment by aidan-casey** [comment]
With [`FrameworkIntegrationTestCase`](https://github.com/tempestphp/tempest-framework/blob/main/tests/Integration/FrameworkIntegrationTestCase.php), the SQLite database is deleted and re-copied _every_ test run. Nothing is currently happening with other dialects like Laravel supports with the `RefreshDatasebase` trait.

I think our default stance should be to _not_ refresh the database on every test case, as that will impact performance. Obviously we do need to support this though, since you will need to do it for tests around migrations, etc.

This probably goes hand-in-hand with a refactor of the test helpers.

**Comment by Treggats** [comment]
> I think our default stance should be to not refresh the database on every test case

I agree, only on the parts where the tables are being recreated. what data is stored inside them is of less concern I gather

**Comment by brendt** [comment]
Gotcha! Ok let's take this into account for alpha2 :)

**Comment by brendt** [comment]
I'm gonna look into this

**Comment by brendt** [comment]
Rollback in itself works, although there are still several errors on MySQL and Postgres. Gonna make a new issue for it
